Foundation/FileManager+Win32.swift

```diff
@@ -312,7 +312,20 @@ extension FileManager {
                               nil)
         }
         if hFile == INVALID_HANDLE_VALUE {
-            throw _NSErrorWithWindowsError(GetLastError(), reading: true)
+            let error = GetLastError()
+            if error == ERROR_FILE_NOT_FOUND || error == ERROR_PATH_NOT_FOUND {
+                return try path.withCString(encodedAs: UTF16.self) {
+                    let dwLength = GetFullPathNameW($0, 0, nil, nil)
+                    var szPath = Array<WCHAR>(repeating: 0, count: Int(dwLength + 1))
+                    let result = GetFullPathNameW($0, dwLength + 1, &szPath, nil)
+                    guard result != 0 && result <= dwLength + 1 else {
```

Comment:
I think that `result <= szPath.count` is nicer, in fact, at that point you could change `let result` into `dwLength` and get something like:

`guard dwLength > 0 && dwLength <= szPath.count`

which feels more obvious